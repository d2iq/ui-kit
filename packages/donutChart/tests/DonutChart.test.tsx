import React from "react";
import { shallow } from "enzyme";
import serializer from "@emotion/jest";
import toJson from "enzyme-to-json";

import { DonutChart } from "../";

expect.addSnapshotSerializer(serializer);

describe("DonutChart", () => {
  it("renders default", () => {
    const component = shallow(
      <DonutChart
        data={[
          {
            percentage: 10,
            color: "#F00"
          }
        ]}
      />
    );

    expect(toJson(component)).toMatchSnapshot();
  });
  it("renders with text and label", () => {
    const component = shallow(
      <DonutChart
        data={[
          {
            percentage: 10,
            color: "#F00"
          }
        ]}
        label="10%"
        text="1 of 10"
      />
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
