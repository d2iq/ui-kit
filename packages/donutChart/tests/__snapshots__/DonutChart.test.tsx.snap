// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonutChart renders default 1`] = `
<svg
  data-cy="donutChart"
  viewBox="0 0 33.33098861837907 33.33098861837907"
>
  <circle
    cx={16.665494309189533}
    cy={16.665494309189533}
    fill="transparent"
    r={15.915494309189533}
    strokeWidth={1.5}
    style={
      Object {
        "stroke": "rgba(218,221,226,0.65)",
      }
    }
  />
  <circle
    cx={16.665494309189533}
    cy={16.665494309189533}
    fill="none"
    key="0"
    r={15.915494309189533}
    strokeDasharray="10 90"
    strokeDashoffset={125}
    strokeWidth={1.5}
    style={
      Object {
        "stroke": "#F00",
      }
    }
  />
  <text
    style={
      Object {
        "textAnchor": "middle",
      }
    }
    x={16.665494309189533}
    y={16.665494309189533}
  >
    <tspan
      data-cy="donutChart-label"
      dominantBaseline="central"
      fontSize={6}
    />
  </text>
</svg>
`;

exports[`DonutChart renders with text and label 1`] = `
.emotion-0 {
  color: var(--themeTextColorSecondary, #76797E);
  fill: var(--themeTextColorSecondary, #76797E);
}

.inverseColorMode .emotion-0,
.emotion-0.inverseColorMode {
  color: var(--themeTextColorSecondaryInverted, #AEB0B4);
  fill: var(--themeTextColorSecondaryInverted, #AEB0B4);
}

<svg
  data-cy="donutChart"
  viewBox="0 0 33.33098861837907 33.33098861837907"
>
  <circle
    cx={16.665494309189533}
    cy={16.665494309189533}
    fill="transparent"
    r={15.915494309189533}
    strokeWidth={1.5}
    style={
      Object {
        "stroke": "rgba(218,221,226,0.65)",
      }
    }
  />
  <circle
    cx={16.665494309189533}
    cy={16.665494309189533}
    fill="none"
    key="0"
    r={15.915494309189533}
    strokeDasharray="10 90"
    strokeDashoffset={125}
    strokeWidth={1.5}
    style={
      Object {
        "stroke": "#F00",
      }
    }
  />
  <text
    style={
      Object {
        "textAnchor": "middle",
      }
    }
    x={16.665494309189533}
    y={16.665494309189533}
  >
    <tspan
      data-cy="donutChart-label"
      dominantBaseline="unset"
      fontSize={6}
    >
      10%
    </tspan>
    <tspan
      className="emotion-0"
      data-cy="donutChart-text"
      fontSize={3}
      x={16.665494309189533}
      y={20.865494309189533}
    >
      1 of 10
    </tspan>
  </text>
</svg>
`;
